packages/service-library/src/servicelib/osparc_generic_resource.py

```diff
@@ -0,0 +1,112 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+from servicelib.logging_utils import log_catch, log_context
+
+_logger = logging.getLogger(__name__)
+
+Ident = TypeVar("Ident")
+Res = TypeVar("Res")
+
+
+class BaseOsparcGenericResourceManager(ABC, Generic[Ident, Res]):
```

Comment:
Since you are only using this in director-v2, why to put it in `servicelib`? 

I know you see this as a generic approach (the name cannot be more generic) but in my experience we end up having many of these solutions around and only used once.

My suggestion is to keep this as an utility in the director-v2 and when there are many use-cases that can be solved using this pattern, then we can move it to `servicelib`

In my view, this is a convenience wrapper for basically handle
-  `safe_remove` 
-  `get_or_create` 
of "something" (which you call resource). These functions are build based on some pre-definitions `get`, `create` and `destroy` of that "something".

This can be indeed useful but I would keep it in a reduced context.